buildpacktoml_test.go

```diff
@@ -0,0 +1,745 @@
+								Version: "some-version-orig", // top-level version is deprecated in buildpack API 0.3
+							},
+							{
+								Name:     "some-dep",
+								Metadata: map[string]interface{}{"version": "v1"},
+							},
+							{
+								Name:     "some-replace-version-dep",
+								Metadata: map[string]interface{}{"version": "some-version-orig"},
+							},
+							{
+								Name: "some-unmet-dep",
+							},
+						},
+					}
+
+					mkfile(t,
+						"[[bom]]\n"+
+							`name = "some-deprecated-bp-replace-version-dep"`+"\n"+
+							"[bom.metadata]\n"+
+							`version = "some-version-new"`+"\n"+
+							"[[bom]]\n"+
+							`name = "some-dep"`+"\n"+
+							"[bom.metadata]\n"+
+							`version = "v1"`+"\n"+
+							"[[bom]]\n"+
+							`name = "some-replace-version-dep"`+"\n"+
+							"[bom.metadata]\n"+
+							`version = "some-version-new"`+"\n",
+						filepath.Join(appDir, "launch-A-v1.toml"),
+					)
+
+					mkfile(t,
+						"[[unmet]]\n"+
+							`name = "some-unmet-dep"`+"\n",
+						filepath.Join(appDir, "build-A-v1.toml"),
+					)
+
+					br, err := bpTOML.Build(bpPlan, config)
+					if err != nil {
+						t.Fatalf("Unexpected error:\n%s\n", err)
+					}
+
+					if s := cmp.Diff(br, lifecycle.BuildResult{
+						BOM: []lifecycle.BOMEntry{
+							{
+								Require: lifecycle.Require{
+									Name:     "some-deprecated-bp-replace-version-dep",
+									Metadata: map[string]interface{}{"version": "some-version-new"},
+								},
+								Buildpack: lifecycle.GroupBuildpack{ID: "A", Version: "v1"}, // no api, no homepage
+							},
+							{
+								Require: lifecycle.Require{
+									Name:     "some-dep",
+									Metadata: map[string]interface{}{"version": "v1"},
+								},
+								Buildpack: lifecycle.GroupBuildpack{ID: "A", Version: "v1"}, // no api, no homepage
+							},
+							{
+								Require: lifecycle.Require{
+									Name:     "some-replace-version-dep",
+									Metadata: map[string]interface{}{"version": "some-version-new"},
+								},
+								Buildpack: lifecycle.GroupBuildpack{ID: "A", Version: "v1"}, // no api, no homepage
+							},
+						},
+						Labels:      []lifecycle.Label{},
+						MetRequires: []string{"some-deprecated-bp-replace-version-dep", "some-dep", "some-replace-version-dep"},
+						Processes:   []launch.Process{},
+						Slices:      []layers.Slice{},
+					}); s != "" {
+						t.Fatalf("Unexpected:\n%s\n", s)
+					}
+				})
+
+				it("should include labels", func() {
+					mkfile(t,
+						"[[labels]]\n"+
+							`key = "some-key"`+"\n"+
+							`value = "some-value"`+"\n"+
+							"[[labels]]\n"+
+							`key = "some-other-key"`+"\n"+
+							`value = "some-other-value"`+"\n",
+						filepath.Join(appDir, "launch-A-v1.toml"),
+					)
+
+					br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+					if err != nil {
+						t.Fatalf("Unexpected error:\n%s\n", err)
+					}
+
+					if s := cmp.Diff(br, lifecycle.BuildResult{
+						BOM: nil,
```

Comment:
It feels a little weird to be returning `nil` here instead of an empty array (and also for `MetRequires`). But this is somewhat consistent with the behavior when there is no launch.toml - we bail out early without ever setting `Labels`, `Processes`, or `Slices`.

Reply:
On thinking about this a little bit more, this is not the behavior we had before (before the refactor, when everything was in one function). We used to just move on to processing the next buildpack. I think we should initialize all these fields to valid values.